velox/exec/Spill.h

```diff
@@ -0,0 +1,350 @@
+    index_ = 0;
+    numRowsInVector_ = rowVector_->size();
+  }
+
+ private:
+  std::string path_;
+  // Byte size of the backing file. Set when finishing writing.
+  uint64_t fileSize_ = 0;
+  std::unique_ptr<WriteFile> output_;
+  std::unique_ptr<SpillInput> input_;
+};
+
+// Describes a bit range inside a 64 bit hash number for use in
+// partitioning data over multiple sets of spill files.
+struct HashBitRange {
+  // Low bit number of hash number bit range
+  uint8_t begin;
+  // Bit number of first bit above the hash number bit range.
+  uint8_t end;
+};
+
+// Sequence of files for one partition of the spilled data. If data is
+// sorted, each file is sorted. The globally sorted order is produced
+// by merging the constituent files.
+class SpillFileList {
+ public:
+  // Constructs a set of spill files. 'type' is a RowType describing the
+  // content. 'path' is a file path prefix. 'targetBatchSize is the target size
+  // of a single RowVector in rows. 'targetFileSize' is the target byte size of
+  // a single file in the file set. 'pool' and 'mappedMemory' are used for
+  // buffering and constructing the result data read from 'this'.
+  SpillFileList(
+      RowTypePtr type,
+      const std::string path,
+      uint64_t targetBatchSize,
+      uint64_t targetFileSize,
+      memory::MemoryPool& pool,
+      memory::MappedMemory& mappedMemory)
+      : type_(type),
+        path_(path),
+        targetBatchSize_(targetBatchSize),
+        targetFileSize_(targetFileSize),
+        pool_(pool),
+        mappedMemory_(mappedMemory) {}
+
+  // Adds 'rows' for the positions in 'indices' into 'this'. The indices
+  // must produce a view where the rows are sorted if sorting is desired.
+  // Consecutive calls must have sorted data so that the first row of the
+  // next call is not less than the last row of the previous call.
+  void write(RowVectorPtr rows, const folly::Range<IndexRange*> indices);
+  std::vector<std::unique_ptr<SpillFile>> files() {
+    return std::move(files_);
+  }
+
+  // Closes one file of output. Subsequent calls to 'write' start a
+  // different sorted run and must be ordered between themselves but not
+  // with respect to calls before flush(). If 'close' is true, write must
+  // not be called and the file set can only be read.
+  void flush(bool close);
+
+ private:
+  const RowTypePtr type_;
+  const std::string path_;
+  const uint64_t targetBatchSize_;
+  const uint64_t targetFileSize_;
+  memory::MemoryPool& pool_;
+  memory::MappedMemory& mappedMemory_;
+  std::unique_ptr<VectorStreamGroup> batch_;
+  bool isOpen_ = false;
+  std::vector<std::unique_ptr<SpillFile>> files_;
+};
+
+// Represents all spilled data of an operator, e.g. order by or group
+// by. This has one SpillFileList per partition of spill data.
+class SpillState {
+ public:
+  // Constructs a SpillState. 'type' is the content RowType. 'path' is the file
+  // system path prefix. 'bits' is the hash bit field for partitioning data
```

Comment:
"hash bit field" is confusing; perhaps, update to clarify that this represents a range of bits